### PR TITLE
Clean up interface diffusion

### LINKED ISSUE #11765: Clean up InterfaceDiffusion in test suite
## Rationale

There have been 1 billion questions about `InterfaceDiffusion` on the user list recently, and there's some misunderstanding about how it works

## Description

The way the residuals are organized is confusing. Moreover, we could create two different `InterfaceDiffusion` kernels (credit @YaqiWang for the "penalty" version) that converge to the same answer (at different rates) that demonstrate different implementations

## Impact

Reduce confusion among users

### PR DESCRIPTION
**Pull request by lindsayad**
- Add new `PenaltyInterfaceDiffusion` test object; this object displays second order convergence rates for first order LAGRANGE elements
- Modify existing `InterfaceDiffusion` object so that its more clear what it's trying to do; note that this object combined with a `MatchedValueBC` is convergent but not optimal (e.g. it does not display second order convergence for first order LAGRANGE)
- Update most tests to use the `PenaltyInterfaceDiffusion` object
- Make `1d_interface` test get properties from volume and neighbor materials instead of boundary material. This is more like reality

Closes #11765

### CODE REVIEW

File: test/include/interfacekernels/PenaltyInterfaceDiffusion.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * DG kernel for interfacing diffusion between two variables on adjacent blocks
+ */
+class PenaltyInterfaceDiffusion : public InterfaceKernel
+{
+public:
+  PenaltyInterfaceDiffusion(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual(Moose::DGResidualType type);
```
Comment (permcody) [inline test/include/interfacekernels/PenaltyInterfaceDiffusion.h]: `override`

**Review by permcody — CHANGES_REQUESTED** [review]
Looks good, just add the override keyword in your new test class.

**Comment by mlesueur** [comment]
Thanks for clarifying the way to implement the InterfaceKernels @lindsayad . Thanks to that we managed to fix the jacobians of our custom InterfaceKernel and improve the convergence!

**Comment by lindsayad** [comment]
Great to hear!

**Review by permcody — APPROVED** [review]

**Pull request merged**